### ISSUE TITLE
[Improvement]: Add ImGui.SeparatorText widget to lua

### ISSUE THREAD

**Issue by Harmless05** [Yimura/YimMenu#2758]
### Which feature

ImGui::SeparatorText for YimMenu Lua

### Reason

Perhaps we could consider adding SeparatorText to YimMenu Lua? It's already being used in YimMenu for 'titles', and it would make it much simpler for Lua script developers to add their own titles for scripts as well.

### Additional context

How the 'title' looks like.

![image](https://github.com/YimMenu/YimMenu/assets/88204341/ffc9e787-3391-419b-a1f2-f6bf36d0fa17)

**Comment by Vitamin-X** [comment]
there is also no component.sub_title

Yim Itself can do something like that

`ImGui::BeginGroup();
components::sub_title("TOXIC"_T);`

and same with get_listbox_dimensions, from here
 `ImGui::BeginListBox("##toxic", get_listbox_dimensions()))`
 
 no clue how to set it with lua

**Comment by Harmless05** [comment]
Another great addition would be `ImGui::IsKeyPressed(key)` for creating a simpler hotkey system in Lua.

**Comment by pierrelasse** [comment]
> Another great addition would be `ImGui::IsKeyPressed(key)` for creating a simpler hotkey system in Lua.

https://github.com/YimMenu/YimMenu/issues/2335

**Comment by malloc2k** [comment]
I don’t know if you guys realise and it’s not a really great of a fix but if you really wanna see this on the menu you can edit that yourselves but the problem is for every update you’d have to redo it all the time… 
Just letting y’all know ;)

**Comment by Harmless05** [comment]
> I don’t know if you guys realise and it’s not a really great of a fix but if you really wanna see this on the menu you can edit that yourselves but the problem is for every update you’d have to redo it all the time… Just letting y’all know ;)

Do you mean creating my own (custom) version of YimMenu or what exactly?

**Comment by xiaoxiao921** [comment]
> Another great addition would be `ImGui::IsKeyPressed(key)` for creating a simpler hotkey system in Lua.

https://github.com/search?q=repo%3Areturn-of-modding%2FReturnOfModding%20%22IsKeyPressed%22&type=code

This should be pretty much copy pastable, feel free to open a PR, I'm not doing it myself cause I've got other things going currently.

For the OP the exact same kind of binding can be done in a PR too, feel free to open one i'll also happily review it

**Comment by Harmless05** [comment]
I don't think it would be that hard to add this `ImGui::SeparatorText` to YimMenu, since it is already used (and the other stuff).

#2335 Would also like some "missing" Lua API features.

**Comment by xesdoog** [comment]
> Another great addition would be `ImGui::IsKeyPressed(key)` for creating a simpler hotkey system in Lua.

Absolutely agree
